### PR TITLE
nodePackages.postcss-cli: fix execution of binary

### PR DESCRIPTION
**Pull request by Luflosi**
###### Motivation for this change
Without this change, executing the postcss binary results in the following error:
Error: Cannot find module 'postcss'

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/node-packages/default.nix [hunk 1]
```diff
@@ -202,6 +202,15 @@ let
       '';
     };
 
+    postcss-cli = super.postcss-cli.override {
+      nativeBuildInputs = [ pkgs.makeWrapper ];
+      postInstall = ''
+        wrapProgram "$out/bin/postcss" \
+        --prefix NODE_PATH : ${self.postcss}/lib/node_modules
```
Comment (roberth) [inline pkgs/development/node-packages/default.nix:209]:
```suggestion
          --prefix NODE_PATH : ${self.postcss}/lib/node_modules
```

**Review by roberth — COMMENTED** [review]
LGTM + nitpick.

**Comment by roberth** [comment]
@GrahamcOfBorg build nodePackages.postcss-cli

Perhaps a passthru.tests would be nice.

File: pkgs/development/node-packages/default.nix (hunk 1 above)
Comment (Luflosi) [reply pkgs/development/node-packages/default.nix]: Thanks, fixed.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130027` at 87f17088 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>lumo</li>
  </ul>
</details>
<details>
  <summary>18 packages built successfully:</summary>
  <ul>
    <li>parity-ui</li>
    <li>python38Packages.batchspawner</li>
    <li>python38Packages.dockerspawner</li>
    <li>python38Packages.jupyterhub</li>
    <li>python38Packages.jupyterhub-ldapauthenticator</li>
    <li>python38Packages.jupyterhub-systemdspawner</li>
    <li>python38Packages.jupyterhub-tmpauthenticator</li>
    <li>python38Packages.oauthenticator</li>
    <li>python39Packages.batchspawner</li>
    <li>python39Packages.dockerspawner</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.jupyterhub-ldapauthenticator</li>
    <li>python39Packages.jupyterhub-systemdspawner</li>
    <li>python39Packages.jupyterhub-tmpauthenticator</li>
    <li>python39Packages.oauthenticator</li>
    <li>teleprompter</li>
    <li>vscode-extensions.matklad.rust-analyzer</li>
    <li>vscode-extensions.vadimcn.vscode-lldb</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Merging this before it has a merge conflict. We can add the passthru tests later.

**Comment by Luflosi** [comment]
How can I add such a test? I'm having trouble finding a good example in Nixpkgs. I don't think any Node packages have any tests.

**Comment by roberth** [comment]
The manual [has a section](https://nixos.org/manual/nixpkgs/stable/#ssec-package-tests-writing). It's somewhat of a new thing to write this kind of test.

Tests in `checkPhase` are more prevalent, but those don't run in a clean environment, which I think is quite important because of the presence of node modules in the build environment, which is not realistic.

**Comment by Luflosi** [comment]
I saw that section. I could probably successfully write a test for a "normal" package.
However this is a Node package, where the list of packages is in `pkgs/development/node-packages/node-packages.json`, the automatically generated package definitions are in `pkgs/development/node-packages/node-packages.nix` and my override is in `pkgs/development/node-packages/default.nix`. I could probably add
```nix
passthru.tests = {
  simple-execution = callPackage ./tests.nix { };
};
```
to the override but where would I put `test.nix`? Putting it in `pkgs/development/node-packages/` seems very wrong as there are no other tests there. I could probably inline the tests but I don't think it's done like that anywhere else, is it?

**Comment by roberth** [comment]
You could definitely add the `passthru.tests` attribute in the overrides file.
I'm never hard to please when someone wants to write a test and you're the first one to do it for a node package, so more power to you.
If I were doing it, I'd use `callPackage` and put the file at a path like `pkgs/development/node-packages/package-tests/postcss-cli.nix`.
I've written this kind of test inline before, but only when it was one package per file.

**Comment by Luflosi** [comment]
Thank you for your suggestion. I created #130204 to add a basic test.

**Pull request merged**